pkg/runtime/runtime.go

```diff
@@ -94,7 +96,14 @@ func NewRuntime() {
 		panic(err.Error())
 	}
 
-	Provider = &provider{coe: cluster}
+	var c ExtractedCredential
+	if extCreds == nil {
```

Comment:
Can you explain why we should have this if/else?

Reply:
if nil we will use the default extracted credentials. This should be in the godoc for the function and will add now.

Reply:
I want to clarify my last comment.  The runtime interface is structured so that if you call ```NewRuntime``` the Runtime pkg will identify what provider you are using.   If we add ```ExtractedCredential``` type to the ```NewRuntime``` call, ```pkg/app/app.go``` needs to know what runtime provider is being used in order to provide a working ```ExtractedCredential``` type.  The reason for that is because ```ExtractedCredential``` is stored in a cluster specific resource, a secret.  My point is I don't think we want to pass ```ExtractedCredential``` as a param to ```NewRuntime```, rather we should add it behind the runtime interface it be initialized after the ```NewRuntime``` call.